### ISSUE TITLE
device id is not passed to Azure AD with MS Intune (error we are getting error Set Up your device to get access)

### ISSUE THREAD

**Issue by nagendra-devireddy** [wrobins/cordova-plugin-msal#80]
Hi All,

I'm having some issues trying to authenticate my users via the Microsoft Intune app. The problem is that even though the users have complaint device on the Intune app, every time that they try to sign in they are still prompt to install the Intune app as if the app was completely unable to communicate with Intune and verify that the user actually has all the permissions required to go through.

We have verified back end azure logs, device id is missing in the request & it is hitting the conditional access policy and making the sign-in/device un-trusted device.

Please find configurations as below
{
  authorities: [
    {
      type: 'AAD',
      audience: 'AzureADMyOrg',
      authorityUrl: 'https://login.microsoftonline.com/TENENT_ID',
      default: true
    }
  ],
  scopes: ['CLIENT_ID/.default'],
  authorizationUserAgent: 'WEBVIEW'
}

![image](https://user-images.githubusercontent.com/5219581/122341718-6cdafb80-cf61-11eb-8693-b35e5b424c40.png)

Thanks in Advance !!

**Comment by trpraveenreddy** [comment]
Hi @nagendra-devireddy  Did you find any solution for this? We are running into similar issue? Thank you

**Comment by nagendra-devireddy** [comment]
Nope. we are planning to use https://ionic.io/products/auth-connect

**Comment by muditpant-ally** [comment]
Hi @nagendra-devireddy did using https://ionic.io/products/auth-connect solved your issue ?

**Comment by jeffortdev** [comment]
I think you need to use brokered authentication for the plugin to pass the deviceID.

**Comment by bradical1379** [comment]
Try setting your `authorityUrl` to `''` since you are using `AAD`.

https://github.com/wrobins/cordova-plugin-msal#authorityurl
